lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2399,10 +2408,28 @@ LogicalResult StmtEmitter::visitStmt(InstanceOp op) {
 
     // Emit the value as an expression.
     ops.clear();
-    // output ports were lowered to wire
-    if (elt.isOutput())
+    // output ports that are not connected to single use output ports were lowered to wire
+    OutputOp output = dyn_cast_or_null<OutputOp>(portVal.getUses().begin()->getOwner());
+    if (elt.isOutput() && portVal.hasOneUse() && output) {
+      auto module = output->getParentOfType<HWModuleOp>();
+      SmallVector<ModulePortInfo> portInfo = getModulePortInfo(module);
+
+      for (size_t portNum = 0, operandNum = 0; portNum < portInfo.size(); ++portNum) {
```

Comment:
I think you can use getModuleResultNameAttr here.

Reply:
How do you get the operand number for the second argument of `getModuleResultNameAttr`?

Reply:
I think that portVal is guaranteed to be a `OpResult`, which has a `getResultNumber()` method.  Does that work?

Reply:
I agree with darth that computing the number from the IR would be better.

Reply:
Oh, apparently it starts at 1 instead of at 0. :/

Reply:
Okay it's still not working...

Reply:
Okay I fixed it!